worlds/kh1/__init__.py

```diff
@@ -0,0 +1,300 @@
+                if self.options.puppies == "full" and name == "All Puppies":
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "Atlantica":
+                if self.options.atlantica:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "Mermaid Kick":
+                if self.options.atlantica:
+                    if self.options.extra_shared_abilities:
+                        item_pool += [self.create_item(name) for _ in range(0, 2)]
+                    else:
+                        item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "Crystal Trident":
+                if self.options.atlantica:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "High Jump":
+                if self.options.extra_shared_abilities:
+                    item_pool += [self.create_item(name) for _ in range(0, 3)]
+                else:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "Progressive Glide":
+                if self.options.extra_shared_abilities:
+                    item_pool += [self.create_item(name) for _ in range(0, 4)]
+                else:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "End of the World":
+                if self.options.end_of_the_world_unlock.current_key == "item":
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name == "EXP Zero":
+                if self.options.exp_zero_in_pool:
+                    item_pool += [self.create_item(name) for _ in range(0, quantity)]
+            elif name not in prefilled_items:
+                item_pool += [self.create_item(name) for _ in range(0, quantity)]
+        
+        for i in range(self.determine_reports_in_pool()):
+            item_pool += [self.create_item("Ansem's Report " + str(i+1))]
+        
+        while len(item_pool) > total_locations:
+            item_pool.pop(0)
+        
+        while len(item_pool) < total_locations and len(level_up_item_pool) > 0:
+            item_pool += [self.create_item(level_up_item_pool.pop())]
+        
+        # Fill any empty locations with filler items.
+        item_names = []
+        attempts = 0  # If we ever try to add items 200 times, and all the items are used up, lets clear the item_names array, we probably don't have enough items
+        while len(item_pool) < total_locations:
+            item_name = self.get_filler_item_name()
+            if item_name not in item_names:
+                item_names.append(item_name)
+                item_pool.append(self.create_item(item_name))
+                attempts = 0
+            elif attempts >= 200:
+                item_names = []
+                attempts = 0
```

Comment:
This *mostly* resets the loop, but the items were still already added to `item_pool`, which carries over into the "next attempt". This seems unintentional

Reply:
But also, do you have no repeatable filler items?

Keep in mind that `get_filler_item_name()` might be called with item links or `panic_method: "start_inventory"` and new filler items are created "without your input", so if there is a problem with these filler items having multiple copies, then you need to handle that **in** get_filler_item_name and keep track of created filler items on the world class.

Anyway, you'll really want some sort of repeatable filler item I feel like

Reply:
Also, I'd much rather this code not rely on random chance, and instead do something like, check whether enough filler items exist *first*, then do `world.random.choices()`